src/utils/include/utils/quaternion.hpp

```diff
@@ -0,0 +1,136 @@
+/*
```

Comment:
should this file be renamed `Quaternion.hpp`?

Reply:
i think we should not use capital letters in filenames at all... that lead to issues on case-insensitive file systems on windows or mac